offline/serviceWorker.js

```diff
@@ -0,0 +1,95 @@
+/* eslint no-restricted-globals: 0 */
+
+const VERSION = 'jsheroesio-cache-v1';
+
+// Application Shell
+const urlsToCache = [
```

Comment:
Other images/resources will be cached. It's a cache-first strategy, it will attempt to return a resource from the cache, and if it's not there it will fetch from the network and add it to the cache => on subsequent requests it can be served from cache. I got images in offline mode, but i did experience some cases where they weren't yet cached. I'll investigate.

Reply:
thanks, I'll also try the now deploy a bit later

Reply:
Investigated and added a few more comments in order to clarify the flow of events and what the code does.

#### 1. Page loaded (online):
* First event to fire: *INSTALL* - here we cache basic stuff (Application Shell) - because nextjs generates hashed versions for urls, and i don't have access to that in the sw file, i was limited in what i could add in `urlsToCache`.
* Second event to fire: *ACTIVATE* - if it is decided at some point to modify `const VERSION = 'jsheroesio-cache-v1'`, the callback for this event will take care of deleting the old cache.

#### 1a. If the page is loaded in offline mode at this point only `urlsToCache` are cached => due to how the hero image is rendered conditionally it is not shown.

#### 2. Page refreshed / loaded a second time (online):
* Service worker is already installed so the event fired is *FETCH* - we apply caching strategies "network only" for ajax requests/external resources and "cache-first: stale-while-revalidate" for the rest. This strategy means that if there's a cached version available, serve it => ⚡️faster load time ⚡️ , but also fetch a fresh update from the network for next time and add it to the cache.

#### 3. Page refreshed / loaded a third time (offline): 
* All images are displayed as they, along with other resources, were added to the cache in the FETCH event
* Everything works except external requests like analytics, maps, fb stuff

Reply:
awesome work! thanks again